packages/components/src/number-control/index.tsx

```diff
@@ -74,10 +81,12 @@ export function NumberControl(
 			type === inputControlActionTypes.PRESS_UP ||
 			type === inputControlActionTypes.PRESS_DOWN
 		) {
+			// @ts-expect-error TODO: Investigate if this is wrong
```

Comment:
This can be fixed by casting `event` to be a keyboard event (afterall, the action type is "press up" or "press down")

```tsx
const enableShift =
	( event as KeyboardEvent | undefined )?.shiftKey &&
	isShiftStepEnabled;
```